go-runtime/encoding/encoding.go

```diff
@@ -356,6 +375,71 @@ func decodeMap(d *json.Decoder, v reflect.Value) error {
 	return err
 }
 
+func decodeSumType(ctx context.Context, d *json.Decoder, v reflect.Value) error {
+	tr, ok := typeregistry.FromContext(ctx).Get()
+	if !ok {
+		return fmt.Errorf("no type registry found in context")
+	}
+
+	if err := expectDelim(d, '{'); err != nil {
```

Comment:
It should be possible to use `json.Decoder.Decode` directly here no, and avoid all this manual decoding? eg. something like [this](https://go.dev/play/p/EgA1MXlH35C)

Reply:
totally 🤦‍♀️